test/suites/nodeclaim/nodeclaim_test.go

```diff
@@ -343,11 +343,73 @@ var _ = Describe("StandaloneNodeClaim", func() {
 			temp := &karpv1.NodeClaim{}
 			g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(nodeClaim), temp)).To(Succeed())
 			g.Expect(temp.StatusConditions().Get(karpv1.ConditionTypeLaunched).IsTrue()).To(BeTrue())
-			g.Expect(temp.StatusConditions().Get(karpv1.ConditionTypeRegistered).IsFalse()).To(BeTrue())
-			g.Expect(temp.StatusConditions().Get(karpv1.ConditionTypeInitialized).IsFalse()).To(BeTrue())
+			g.Expect(temp.StatusConditions().Get(karpv1.ConditionTypeRegistered).IsUnknown()).To(BeTrue())
+			g.Expect(temp.StatusConditions().Get(karpv1.ConditionTypeInitialized).IsUnknown()).To(BeTrue())
 		}).Should(Succeed())
 
 		// Expect that the nodeClaim is eventually de-provisioned due to the registration timeout
 		env.EventuallyExpectNotFoundAssertion(nodeClaim).WithTimeout(time.Minute * 20).Should(Succeed())
 	})
+	It("should delete a NodeClaim if it references a NodeClass that doesn't exist", func() {
+		nodeClaim := test.NodeClaim(karpv1.NodeClaim{
+			Spec: karpv1.NodeClaimSpec{
+				Requirements: []karpv1.NodeSelectorRequirementWithMinValues{
+					{
+						NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+							Key:      v1.LabelInstanceCategory,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{"c"},
+						},
+					},
+					{
+						NodeSelectorRequirement: corev1.NodeSelectorRequirement{
+							Key:      karpv1.CapacityTypeLabelKey,
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{karpv1.CapacityTypeOnDemand},
+						},
+					},
+				},
+				NodeClassRef: &karpv1.NodeClassReference{
+					Group: object.GVK(nodeClass).Group,
+					Kind:  object.GVK(nodeClass).Kind,
+					Name:  nodeClass.Name,
+				},
+			},
+		})
+		// Don't create the NodeClass and expect that the NodeClaim fails and gets deleted
+		env.ExpectCreated(nodeClaim)
+		env.ExpectDeleted(nodeClaim)
```

Comment:
This does the actual delete. Don't we just want to check if this is not found? Is this testing garbage collection or termination?